### ISSUE TITLE
Upgrading from 0.x to 1.x fails to refresh access token when remember device enabled

### ISSUE THREAD

**Issue by karl-d** [aws-amplify/amplify-flutter#3995]
### Description

We have an issue post migrating from 0.6.14 -> 1.4.1 with users being logged out of our mobile application.

After they update our mobile application they are having an issue with the refresh token being incorrect when trying to get a new access token. They receive the following response from Cognito, “Invalid Refresh Token.”

We have dug into the issue and it looks like it is due to our configuration to allow users to remember their device and the device secret not being migrated in the upgrade. 

This causes the call to refresh the access token to fail, as Cognito requires the device secret to be passed in the request

`With device tracking, these tokens are linked to a single device. If a refresh token is used on any other device, the call failsWith device tracking, these tokens are linked to a single device. If a refresh token is used on any other device, the call fails`

https://aws.amazon.com/blogs/mobile/tracking-and-remembering-devices-using-amazon-cognito-your-user-pools/ 

### Categories

- [ ] Analytics
- [ ] API (REST)
- [ ] API (GraphQL)
- [X] Auth
- [ ] Authenticator
- [ ] DataStore
- [ ] Notifications (Push)
- [ ] Storage

### Platforms

- [X] iOS
- [X] Android
- [ ] Web
- [ ] macOS
- [ ] Windows
- [ ] Linux

### Flutter Version

3.10.6

### Amplify Flutter Version

1.4.1

### Deployment Method

Custom Pipeline

**Comment by Jordan-Nelson** [comment]
Hello @karl-d - Thanks for opening the issue. I am looking into this.

If you have the full exception / stack trace available, can you share it?

Can you confirm that the end user is logged out and is able to then log in again?

**Comment by karl-d** [comment]
HI @Jordan-Nelson,

I will attempt to get a stack trace for you, we dont have one currently. I do have the below image of the exception that was thrown though.
<img width="974" alt="image" src="https://github.com/aws-amplify/amplify-flutter/assets/11346844/c4cab570-d806-4693-9068-fabe6ac62ef4">

Yes I can confirm that users are able to log in successfully afterwards. We can see the Amplify code deleting the existing tokens after it fails to generate a new access token, using the migrated refresh_token.

**Comment by karl-d** [comment]
Hi @Jordan-Nelson,

I have debugged through and this is the request to get a new refresh token that errors out in fetch_auth_session_state_machine.dart
```
#0      HttpOperation.deserializeOutput (package:smithy/src/http/http_operation.dart:312:7)
<asynchronous suspension>
#1      CancelableCompleter.complete.<anonymous closure> (package:async/src/cancelable_operation.dart:425:16)
<asynchronous suspension>
```

<img width="1786" alt="image" src="https://github.com/aws-amplify/amplify-flutter/assets/11346844/ce03140a-8b19-44fb-babf-4fb988d2bb44">

<img width="725" alt="image" src="https://github.com/aws-amplify/amplify-flutter/assets/11346844/a8b09676-2d03-495e-afe3-3bd344ecef26">

<img width="1213" alt="Screenshot 2023-10-27 at 2 50 27 PM" src="https://github.com/aws-amplify/amplify-flutter/assets/11346844/34ebfe1e-3fe6-4c9d-938d-4ac969dace06">

My suggestion is that in the credential_store_state_machine.dart@_migrateLegacyCredentialStore function should also migrate the device key if there is one present as well.

Please let me know if you require any more information here.

Regards,
Karl

**Comment by Jordan-Nelson** [comment]
@karl-d - I agree that the data should be migrated as well. I am going to mark this as a bug and we will look into a fix. Thanks for taking the time to open an issue.

**Comment by Jordan-Nelson** [comment]
We are working on a fix for this issue. A PR for this is linked above. I will let you know when it has been released.

**Comment by Equartey** [comment]
Hi @karl-d, this has been fixed in `amplify_auth_cognito: ^1.7.1`. 

I'm going to close this issue as complete. Please reach out if you have any issues, thanks.

**Comment by Jordan-Nelson** [comment]
@karl-d - FYI we had to retract v1.7.1 due to an unrelated issue. We will have a new version out as soon as possible that includes this fix.
